sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeDirectoryClient.cs

```diff
@@ -272,244 +273,1054 @@ internal override void SetNameFieldsIfNull()
+            DataLakeRequestConditions sourceConditions = default,
+            DataLakeRequestConditions destinationConditions = default,
+            CancellationToken cancellationToken = default)
+        {
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(Azure)}.{nameof(Storage)}.{nameof(Files)}.{nameof(DataLake)}.{nameof(DataLakeDirectoryClient)}.{nameof(Rename)}");
+
+            try
+            {
+                scope.Start();
+
+                Response<DataLakePathClient> response = await base.RenameAsync(
+                    destinationPath,
+                    sourceConditions,
+                    destinationConditions,
+                    cancellationToken)
+                    .ConfigureAwait(false);
+
+                return Response.FromValue(
+                    new DataLakeDirectoryClient(response.Value.DfsUri, response.Value.Pipeline),
+                    response.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+            finally
+            {
+                scope.Dispose();
+            }
+        }
+        #endregion Rename
+
+        #region Get Access Control
+        /// <summary>
+        /// The <see cref="GetAccessControl"/> operation returns the
+        /// access control data for a path.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/getproperties" />.
+        /// </summary>
+        /// <param name="userPrincipalName">
+        /// Optional.Valid only when Hierarchical Namespace is enabled for the account.If "true",
+        /// the user identity values returned in the x-ms-owner, x-ms-group, and x-ms-acl response
+        /// headers will be transformed from Azure Active Directory Object IDs to User Principal Names.
+        /// If "false", the values will be returned as Azure Active Directory Object IDs.The default
+        /// value is false. Note that group and application Object IDs are not translated because they
+        /// do not have unique friendly names.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on getting the path's access control.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{PathAccessControl}"/> describing the
+        /// path's access control.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+#pragma warning disable AZC0003 // DO make service methods virtual.
```

Comment:
I'm overriding a virtual method, so my understanding is that this method is also virtual.  In addition, you can't mark a method both override and virtual.

Reply:
It's a bug, filed https://github.com/Azure/azure-sdk-tools/issues/291